pkg/app/middlewares/tenant_middleware.go

```diff
@@ -0,0 +1,46 @@
+package middlewares
+
+import (
+	"context"
+
+	"github.com/aserto-dev/aserto-grpc/grpcutil"
+	"github.com/aserto-dev/header"
+	"github.com/aserto-dev/topaz/pkg/cc/config"
+	grpcmiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"google.golang.org/grpc"
+)
+
+type TenantIDIDMiddleware struct {
```

Comment:
A similar suggestion to rename the file to `tenant.go` as the middleware aspect is implied by the directory name.